testcontainers/src/core/logs.rs

```diff
@@ -36,21 +36,19 @@ impl<'d> LogStreamAsync<'d> {
     }
 }
 
-pub(crate) struct LogStream {
-    inner: Box<dyn Read>,
+pub(crate) struct LogStream<R: Read> {
+    inner: R,
 }
 
-impl fmt::Debug for LogStream {
+impl<R: Read> fmt::Debug for LogStream<R> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         f.debug_struct("LogStream").finish()
     }
 }
 
-impl LogStream {
-    pub fn new(stream: impl Read + 'static) -> Self {
-        Self {
-            inner: Box::new(stream),
-        }
+impl<R: Read> LogStream<R> {
+    pub fn new(stream: R) -> Self {
+        Self { inner: stream }
```

Comment:
It's a bit unrelated because I reverted changes of `WaitFor`, but decided to keep it because it removes extra `Box`ing without breaking any backward compatibility

Reply:
I am pretty sure this is a breaking change still. I'd like to keep the boxing to keep the API simpler and the type encapsulated.

Reply:
🤔  looks like it's `pub(crate)` and not exposed. 

I can revert ofc, but seems to be not breaking one